File: Javascript/Authored/Wave.js
function wave(){
  result = [];
  str = str.split("");
  for(b=0;b<str.length;b++){
    if (str[b] !== " "){
      temp = [];
      for(i=0;i<str.length;i++){
        if (i===b){temp.push(str[i].toUpperCase();}else{temp.push(str[i]);}
      }
      result.push(temp.join(""));
    }
  }
  return result
}

letters = "abcd efghi jklmno pqrstu vwxyz".split("");
for(cwtests=0;cwtests<96;cwtests++){
  word = []
  for(wordnum=0;wordnum<Math.floor(Math.random() * 100);wordnum++){
    word.push(letters[Math.floor(Math.random() * (letters.length - 1))]);
  }
  word = word.join("");
  result = wave2(word)
  Test.assertDeepEquals(wave(word), result, "Should return: '"+result+"'");
}

File: Python/Authored/Wave.py
def wave(str):
    result = []
    str = list(str)
    for b in range(0,len(str)):
        if str[b] != " ":
            temp = []
            for i in range(0, len(str)):
                if i == b:
                    temp.append(str[i].upper())
                else:
                    temp.append(str[i])
            result.append("".join(temp))
    return result

import random
letters = list("abcd efghi jklmno pqrstu vwxyz")
for cwtests in range (1,2):
    word = []
    for wordloop in range(0,random.randint(0,200)):
        word.append(letters[random.randint(0,len(letters)-1)])
    word = "".join(word)
    print(word)
